src/Endpoints/TenantToken.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MeiliSearch\Endpoints;
+
+use DateTime;
+use MeiliSearch\Contracts\Endpoint;
+use MeiliSearch\Exceptions\InvalidArgumentException;
+use MeiliSearch\Http\Serialize\Json;
+
+class TenantToken extends Endpoint
+{
+    private function base64url_encode($data)
+    {
+        return rtrim(strtr(base64_encode($data), '+/', '-_'), '=');
+    }
+
+    /**
+     * Generate a new tenant token.
+     *
+     * The $options parameter is an array, and the following keys are accepted:
+     * - apiKey: The API key parent of the token. If you leave it empty the client API Key will be used.
+     * - expiresAt: A DateTime when the key will expire. Note that if an expiresAt value is included it should be in UTC time.
+     */
+    public function generateTenantToken($searchRules, ?array $options = []): string
+    {
+        // Validate every fields
+        if ((!\array_key_exists('apiKey', $options) || '' == $options['apiKey']) && \is_null($this->apiKey)) {
+            throw InvalidArgumentException::emptyArgument('api key');
+        }
+        if (null == $searchRules) {
+            throw InvalidArgumentException::emptyArgument('search rules');
```

Comment:
See [comment here](https://github.com/meilisearch/integration-guides/issues/181#issuecomment-1060399483)

Reply:
I change it for:
```php
        if ((!is_array($searchRules) && !is_object($searchRules)) || null == $searchRules) {
            throw InvalidArgumentException::emptyArgument('search rules');
        }
```
For checking and the type and the fact that `searchRules` exist